kbase-extension/static/kbase/js/widgets/narrative_core/upload/stagingAreaViewer.js

```diff
@@ -439,294 +495,399 @@ define([
-                            if any are checked we leave import button enabled
-                            */
-                            const anyCheckedBoxes = $(
-                                `input.${cssBaseClass}-body__checkbox-input:checked`
+                            const selectUploader = select({}, [
+                                option(),
+                                mappedOptions,
+                                uploaderOptions,
+                            ]);
+                            const downloadBtn = button(
+                                {
+                                    dataDownload: rowData.path,
+                                    class: `${tableBodyCssBaseClass}__button--download`,
+                                    role: 'button',
+                                    ariaLabel: 'Download Data',
+                                    title: 'Download data',
+                                },
+                                span({ class: 'fa fa-download' })
                             );
+                            return div(
+                                {
+                                    class: `${tableBodyCssBaseClass}__select_container`,
+                                },
+                                [selectUploader, downloadBtn, deleteBtn]
+                            );
+                        },
+                    },
+                ],
+                createdRow: function (row, data) {
+                    $(row)
+                        .addClass('kb-staging-table__row')
+                        .attr('data-file-path', data.path)
+                        .attr('data-subdir', data.subdir);
+                },
+                rowCallback: function (row, data) {
+                    const $rowCheckbox = $('td:eq(0)', row).find(
+                        `.${cssBaseClass}-body__checkbox-input`
+                    );
+                    stagingAreaViewer.attachFileIconEvents($('td:eq(1)', row), data);
+                    stagingAreaViewer.attachImportSelectionEvents(
+                        $('td:eq(5)', row),
+                        $rowCheckbox,
+                        data
+                    );
+                    stagingAreaViewer.attachFileNameEvents($('td:eq(2)', row), data);
 
-                            if (!anyCheckedBoxes.length) {
-                                stagingAreaViewer.disableImportButton();
-                            }
-                        }
+                    // attach events to the row checkbox
+                    if (stagingAreaViewer.checkedFiles.has(data.path)) {
+                        $rowCheckbox.prop('checked', true);
                     }
+                    // bind the change event here so it gets triggered
+                    // when selecting a type by the dropdown makes the change
+                    $rowCheckbox.off('click change').on('click change', (e) => {
```

Comment:
It's a little fuzzy when and how elements get added/removed/emptied from the DOM. So we remove any events before rebinding them for the first time.